app/models/comment.rb

```diff
@@ -1,3 +1,10 @@
 class Comment < ApplicationRecord
-  belongs_to :post, class_name: 'Post', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+  belongs_to :user, class_name: 'User', foreign_key: 'user_id'
+
+  after_save :update_comment_counter
+
+  def update_comment_counter
+    post.update(comments_counter: post.comments.count)
```

Comment:
- [ ] **[OPTIONAL]** You can also use the `increment!` method here as demonstrated earlier.